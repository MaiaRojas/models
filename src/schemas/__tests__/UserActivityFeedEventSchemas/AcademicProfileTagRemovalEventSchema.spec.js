const mongoose = require('mongoose/browser');
const UserSchema = require('../../UserSchema')(mongoose);
const { UserActivityFeedEventSchemas: { AcademicProfileTagRemovalEventSchema } } = require('../../')(mongoose);


describe('AcademicProfileTagRemovalEventSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, AcademicProfileTagRemovalEventSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const removedBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);

    const doc = new mongoose.Document({
      removedBy,
      tag: 'someone',
      removalReason: 'someone text',
    }, AcademicProfileTagRemovalEventSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
