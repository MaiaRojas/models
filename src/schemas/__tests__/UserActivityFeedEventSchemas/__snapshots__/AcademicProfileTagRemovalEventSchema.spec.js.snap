// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileTagRemovalEventSchema should fail validation when missing fields are provided 1`] = `
Object {
  "removalReason": [ValidatorError: Path \`removalReason\` is required.],
  "removedBy": [ValidatorError: Path \`removedBy\` is required.],
  "tag": [ValidatorError: Path \`tag\` is required.],
}
`;
